pkg/serverless/serverless.go

```diff
@@ -176,14 +179,17 @@ func callInvocationHandler(daemon *daemon.Daemon, arn string, deadlineMs int64,
 	case <-doneChannel:
 		break
 	}
-	sendSystemEnhancedMetrics(daemon, cpuOffsetErr == nil && uptimeOffsetErr == nil, networkOffsetErr == nil, uptimeOffset, cpuOffsetData, networkOffsetData)
+	sendSystemEnhancedMetrics(daemon, cpuOffsetErr == nil && uptimeOffsetErr == nil, networkOffsetErr == nil, uptimeOffset, cpuOffsetData, networkOffsetData, sendTmpMetrics)
 }
 
-func sendSystemEnhancedMetrics(daemon *daemon.Daemon, emitCPUMetrics, emitNetworkMetrics bool, uptimeOffset float64, cpuOffsetData *proc.CPUData, networkOffsetData *proc.NetworkData) {
+func sendSystemEnhancedMetrics(daemon *daemon.Daemon, emitCPUMetrics, emitNetworkMetrics bool, uptimeOffset float64, cpuOffsetData *proc.CPUData, networkOffsetData *proc.NetworkData, sendTmpMetrics chan bool) {
 	if daemon.MetricAgent == nil {
 		log.Debug("Could not send system enhanced metrics")
 		return
 	}
+
+	sendTmpMetrics <- true
```

Comment:
In the case where the goroutine that consumes from this channel has returned early (ie there was an error when getting the statfs) then this call call will block forever and lead to an invocation timeout.

Reply:
Added a note on the other comment, updated the goroutine that consumes from this channel to not return early

Reply:
Hmmm, one alternative that I hadn't thought of before, instead of sending a `true` into the channel, I think you could alternatively call `close(sendTmpMetrics)` which doesn't block.

Reply:
updated to do it this way!